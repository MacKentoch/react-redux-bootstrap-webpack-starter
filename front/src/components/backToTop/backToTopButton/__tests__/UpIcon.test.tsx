import React from 'react';
import { render } from '@testing-library/react';
import UpIcon from '../UpIcon';

describe('UpIcon component', () => {
  let rootElement: any = null;

  const defaultProps = {
    color: '',
  };

  beforeEach(() => {
    rootElement = document.createElement('div');
    document.body.appendChild(rootElement);
  });

  afterEach(() => {
    rootElement && document.body.removeChild(rootElement);
    rootElement = null;
  });

  it('renders as expected', () => {
    const props = { ...defaultProps };
    const { container } = render(<UpIcon {...props} />, rootElement);
    expect(container.firstChild).toMatchSnapshot();
  });
});
