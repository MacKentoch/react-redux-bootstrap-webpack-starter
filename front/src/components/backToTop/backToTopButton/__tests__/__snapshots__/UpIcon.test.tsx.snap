// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpIcon component renders as expected 1`] = `
<svg
  fill=""
  height="24px"
  viewBox="0 0 512 512"
  width="24px"
>
  <path
    d="M256,213.7L256,213.7L256,213.7l174.2,167.2c4.3,4.2,11.4,4.1,15.8-0.2l30.6-29.9c4.4-4.3,4.5-11.3,0.2-15.5L264.1,131.1
      c-2.2-2.2-5.2-3.2-8.1-3c-3-0.1-5.9,0.9-8.1,3L35.2,335.3c-4.3,4.2-4.2,11.2,0.2,15.5L66,380.7c4.4,4.3,11.5,4.4,15.8,0.2L256,213.7z"
  />
</svg>
`;
